log4j-core/src/main/java/org/apache/logging/log4j/core/util/internal/StringBuilders.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.logging.log4j.core.util.internal;
+
+/**
+ * StringBuilder helpers
+ */
+public class StringBuilders {
+
+    /**
+     * Truncates the content of the given {@code StringBuilder} to the specified maximum number of lines.
+     *
+     * <p>If {@code maxLineCount} is {@code null}, {@link Integer#MAX_VALUE}, or if {@code lineSeparator} is empty,
+     * the method returns without making any changes to the {@code StringBuilder}.
+     *
+     * @param buffer             the {@code StringBuilder} whose content is to be truncated
+     * @param lineSeparator  the line separator used to determine the end of a line
+     * @param maxLineCount   the maximum number of lines to retain in the {@code StringBuilder};
+     *                       if this value is {@code null} or {@link Integer#MAX_VALUE}, no truncation will occur
+     */
+    public static void truncateLines(
```

Comment:
@vy 
Just realized that `truncatLines()` doesn't need `Consumer<StringBuilder> lineSeparatorInjector`, since we are just "cutting" the buffer instead of building new one.

Reply:
@vy If we don't check `maxLineCount == maxLineCount == Integer.MAX_VALUE` for early exit, then even the `depth`(`options.lines`) is default(`Integer.MAX_VALUE`), it will go through the loop.
Or is it better to check if truncate is needed before calling `StringBuilders.truncateLines` in each callers?

Reply:
You're right massage the `if` condition as follows:
```
buffer.length() < delimiter.length() || delimiter.isEmpty() || maxOccurrenceCount == Integer.MAX_VALUE
```